engine/services/engine_service.cc

```diff
@@ -753,27 +772,32 @@ void EngineService::RegisterEngineLibPath() {
     try {
       auto engine_dir_path_res = GetEngineDirPath(engine);
       if (engine_dir_path_res.has_error()) {
-        CTL_ERR(
+        CTL_WRN(
             "Could not get engine dir path: " << engine_dir_path_res.error());
         continue;
       }
       auto engine_dir_path = engine_dir_path_res.value().first;
       auto custom_engine_path = engine_dir_path_res.value().second;
-
-      auto dylib = std::make_unique<cortex_cpp::dylib>(engine_dir_path.string(),
-                                                       "engine");
-
       auto cuda_path = file_manager_utils::GetCudaToolkitPath(ne);
-      // init
-      auto func = dylib->get_function<EngineI*()>("get_engine");
-      auto engine = func();
+
+      // register deps
       std::vector<std::filesystem::path> paths{};
-      auto register_opts = EngineI::RegisterLibraryOption{
-          .paths = paths,
-      };
-      engine->RegisterLibraryPath(register_opts);
-      delete engine;
-      CTL_DBG("Register lib path for: " << engine);
+      paths.push_back(std::move(cuda_path));
+      paths.push_back(std::move(engine_dir_path));
+
+      CTL_DBG("Registering dylib for "
+              << ne << " with " << std::to_string(paths.size()) << " paths.");
+      for (const auto& path : paths) {
+        CTL_DBG("Registering path: " << path.string());
+      }
+
+      auto reg_result = dylib_path_manager_->RegisterPath(ne, paths);
```

Comment:
Seems like we register twice for Windows.
For Linux, I think we don't need to register here, also the LD_LIBRARY_PATH does not work with running process. We have already registered with `server_start_cmd`